api/location.ts

```diff
@@ -77,8 +77,28 @@ export class Location {
       }
 
       return deviceList.reduce((updatedDevices: RingDevice[], data) => {
-        const flatData = flattenDeviceData(data),
-          existingDevice = updatedDevices.find((x) => x.zid === flatData.zid)
+        const flatData = flattenDeviceData(data)
+
+        // Embed component device data in parent device data
```

Comment:
This exists specifically to add additional `RingDeviceData` (i.e. is separate from any kind of HomeKit accessory linking). But it reminded me of Homebridge’s [`addLinkedService`](https://github.com/homebridge/HAP-NodeJS/blob/f3f032547f9c21f61c6338fbaf948b8258415012/src/lib/Service.ts#L234-L240)—I wonder whether the Homebridge accessories for thermostats and temperature sensors should be linked.

Reply:
Copy-pasting @dgreif’s [comment above](https://github.com/dgreif/ring/pull/331#discussion_r429253567):
> I'm hesitant to add the component devices to the flat data of the parent. The downside is you don't have a reference to the `RingDevice` instance for that component device, just to its data. It may be more useful to add a `getComponentDevices(parentZid: string) => RingDevice[]` method to each device. I haven't seen the `addLinkedService` method before but it definitely seems useful in this case. Maybe after all accessories have been created in the platform, we could then go back and do linking.

Reply:
Added a `getComponentDevices` method in https://github.com/dgreif/ring/pull/331/commits/28df3881bbacf257e092b2d4d19d880830d8185b. I haven’t tested it as thoroughly as the previous implementation—I’ll try to do so tomorrow as temperatures fluctuate.